apps/web/vibes/soul/sections/cart/schema.ts

```diff
@@ -14,3 +14,14 @@ export const cartLineItemActionFormDataSchema = z.discriminatedUnion('intent', [
     id: z.string(),
   }),
 ]);
+
+export const couponCodeActionFormDataSchema = z.discriminatedUnion('intent', [
+  z.object({
+    intent: z.literal('apply'),
+    couponCode: z.string({ required_error: 'Please enter a valid promo code' }),
```

Comment:
Not sure how we'll be able to translate these 🤔

Reply:
Make the schema a function that takes the label and returns the schema. I think we've done this for a couple of schemas already.

Reply:
True! Updated